src/Features/CSharp/Portable/Completion/CompletionProviders/FunctionPointerUnmanagedCallingConventionCompletionProvider.cs

```diff
@@ -0,0 +1,139 @@
+                AddTypes(completionItems, contextPosition, semanticModel, cancellationToken);
+
+                // Even if we didn't have types, there are four magic calling conventions recognized regardless.
+                // We add these after doing the type lookup so if we had types we can show that instead
+                foreach (var callingConvention in s_predefinedCallingConventions)
+                {
+                    completionItems.Add(CompletionItem.Create(callingConvention, tags: GlyphTags.GetTags(Glyph.Keyword)));
+                }
+
+                context.AddItems(completionItems);
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
+            {
+                // nop
+            }
+        }
+
+        private static void AddTypes(HashSet<CompletionItem> completionItems, int contextPosition, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            // We have to find the set of types that meet the criteria listed in
+            // https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/function-pointers.md#mapping-the-calling_convention_specifier-to-a-callkind
+            // We skip the check of an type being in the core assembly since that's not really necessary for our work.
+            var compilerServicesNamespace = semanticModel.Compilation.GlobalNamespace.GetQualifiedNamespace("System.Runtime.CompilerServices");
+            if (compilerServicesNamespace == null)
+            {
+                return;
+            }
+
+            foreach (var type in compilerServicesNamespace.GetTypeMembers())
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                if (type.DeclaredAccessibility == Accessibility.Public && type.Name.StartsWith("CallConv"))
+                {
+                    var displayName = type.Name.Substring("CallConv".Length);
+                    completionItems.Add(
+                        SymbolCompletionItem.CreateWithSymbolId(
+                            displayName,
+                            ImmutableArray.Create(type),
+                            rules: CompletionItemRules.Default,
+                            contextPosition));
+                }
+            }
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private class CompletionItemComparer : IEqualityComparer<CompletionItem>
+        {
+            public static readonly IEqualityComparer<CompletionItem> Instance = new CompletionItemComparer();
+
+            public bool Equals(CompletionItem? x, CompletionItem? y)
+            {
+                return x?.DisplayText == y?.DisplayText;
```

Comment:
Is it possible to have two items of the same type name in different sub-namespaces of ```System.Runtime.CompilerServices```, one is imported one not?  Or does ```compilerServicesNamespace.GetTypeMembers()``` only return direct type members?